src/Roslyn.Suppressions/Roslyn.Suppressions/SupportedSuppressionBuilder.tt

```diff
@@ -115,24 +137,43 @@ namespace SonarLint.VisualStudio.Roslyn.Suppressions
                 .OrderBy(x => x);
         }
 
-        private IEnumerable<DiagnosticDescriptor> GetDiagnosticDescriptors(Type t)
+        private IEnumerable<string> GetIdsUsingReflection(Type t)
         {
             try
             {
-                if (!t.IsAbstract && t.GetCustomAttributes(DiagnosticAnalyzerAttribute, true).Any())
+                if (IsDiagnosticAnalyzer(t))
                 {
-                    return Create(t).SupportedDiagnostics;
+                    var analyzer = Create(t);
+                    return GetDiagnosticIds(analyzer);
                 }
             }
             catch(Exception e)
             {
                 WriteLine($"[ERROR] Error creating diagnostic. {t.FullName}, Error: {e.Message}");
             }
-            return Array.Empty<DiagnosticDescriptor>();
+            return Array.Empty<string>();
         }
 
-        private static DiagnosticAnalyzer Create(Type t) =>
-            Activator.CreateInstance(t) as DiagnosticAnalyzer;
+        private static object Create(Type t) =>
+            Activator.CreateInstance(t);
 
-        private static readonly Type DiagnosticAnalyzerAttribute = typeof(DiagnosticAnalyzerAttribute);
+        private bool IsDiagnosticAnalyzer(Type t)
+        {
+            if (t.IsAbstract) { return false; }
+            return t.GetCustomAttributes(true).Any(x => x.GetType().Name == "DiagnosticAnalyzerAttribute");
+        }
+
+        private IEnumerable<string> GetDiagnosticIds(object analyzerInstance)
+        {
+            var propertyInfo = analyzerInstance.GetType().GetProperty("SupportedDiagnostics");
```

Comment:
Fetching the property info every time isn't efficient, but the transformation is so fast it's not worth bothering about.

Also, the transformation is only run when the input files are updated (the input files being any files that are used during the transform i.e. this .tt, and the input binaries. So the transformation will be run once when we update the analyzer version).